### PR TITLE
Document dependency on liboqs-rust Release Managers team

### PR DESCRIPTION
**Pull request by SWilson4**
See https://github.com/open-quantum-safe/tsc/pull/77#discussion_r1780535657. Marked as draft pending input from @thomwiggers on whether or not the name change (liboqs-rust-release-managers -> rust) is necessary.

Unconditionally, changes to `config.yaml` must
- [ ] be approved by 2 members of the OQS TSC
- [ ] not violate permissions documented in GOVERNANCE.md files for sub projects where such files exist

The following goals apply to changes to the file `config.yaml` with exceptions possible, as long as the rationale for the exception is documented by comments in the file:
- [ ] all sub projects should be treated identically wrt roles & responsibilities as per the detailed list below
- [ ] teams/team designations are to be used wherever possible; using personal GH handles should only be used in team definitions
- [ ] Admin changes to the file must be documented by comments as to the rationale of the change

All the following conditions hold for permissions set in `config.yaml`:
-    sub project maintainers have admin rights on the sub projects
-    OQS and sub project release managers have maintainer rights on the sub projects but can themselves set/reset branch protection rules limiting write access to sensitive branches
-    sub project committers have write rights on all branches of the sub projects but can request branch protection rules limiting this
-    sub project contributors (incl. code owners) have write rights on all branches except main on those sub projects
-    OQS and sub project triage actors have triage rights on all branches of the sub projects
-    OQS maintainers and LF admins have admin rights on the organization (e.g., org-wide secret management) as well as maintenance rights on the team configurations

### CODE REVIEW

**Comment by thomwiggers** [comment]
I have added the `liboqs-rust-release-managers` team to Crates.io but adding the comment still seems like a good move.

File: config.yaml
Comment (thomwiggers) [inline config.yaml:227]:
```suggestion
- name: liboqs-rust-release-managers
```

File: config.yaml
Comment (thomwiggers) [inline config.yaml:383]:
```suggestion
    # Note that this team name is relied on by `crates.io`
    liboqs-rust-release-managers: maintain
```

File: config.yaml
Comment (thomwiggers) [inline config.yaml:225]:
```suggestion
# This team is tied to permissions on Crates.io: https://crates.io/teams/github:open-quantum-safe:liboqs-rust-release-managers
```

**Review by thomwiggers — APPROVED** [review]

**Review by dstebila — APPROVED** [review]

**Comment by SWilson4** [comment]
_Technically_ this changes the config.yaml file, even though it doesn't actually change the config. I will play it by the book and wait for another approval from the TSC before merging. (@ashman-p or @bhess or @christianpaquin perhaps?)

**Review by baentsch — APPROVED** [review]

**Pull request merged**
